### PR TITLE
Provide an IdeContextHeader component

### PR DESCRIPTION
**Pull request by chrisbygrave**
Supports CP4I requirement for exposing context for users of multiple instances of the same product

Contributes to #

{{short description}}

#### Changelog

Use the following command to list changes or manually list out new, changed and
removed files.

{{`git diff --name-status $(git merge-base head master)`}}

**New**

- {{new thing}}

**Changed**

- {{change thing}}

**Removed**

- {{removed thing}}

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

##### New production dependencies

- [ ] Verify the dependency has previously been
      [pedigree reviewed](https://pedigree-service.wdc1a.cirrus.ibm.com)
- [ ] Verify the dependency
      [doesn't contain any vulnerabilities](https://snyk.io/vuln)
- [ ] Verify the dependency [bundle size](https://bundlephobia.com) is
      acceptable
- [ ] Verify the dependency is [actively maintained](https://www.npmtrends.com)

### CODE REVIEW

File: packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//
+
+@import '../../globals/scss/carbon-settings';
+@import '../../globals/scss/variables';
+
+.#{$ide-prefix}-context-header {
+  width: 100%;
+  padding: $spacing-03 $spacing-05 7px;
+  background: $ui-background;
+  border-bottom: 1px solid $ui-03;
```
Comment (vladbalanescu) [inline packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss:15]: can we please use a spacing token here?

**Review by vladbalanescu — APPROVED** [review]
Looking awesome, just 1 little tiny detail from me 🙂

File: packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss (hunk 1 above)
Comment (chrisbygrave) [reply packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss:15]: For the 1px or the 7px? I couldn't find a good token to use for 7px that appeased the linter, so I left it as a plain old 7px... happy to use, say, carbon--mini-units(0.875) if that is preferable.

If it's the 1px... I didn't spot that :)

Comment (chrisbygrave) [reply packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss]: @vladbalanescu I've moved the component across to experimental, and also re-worked the CSS there so I could avoid a linter issue. PTAL.

Comment (vladbalanescu) [reply packages/cdai/src/components/IdeContextHeader/_IdeContextHeader.scss]: `carbon--mini-units` is the preferred support here, see [this example](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/a3cf68f2eb36351c8b551385fcc88468a3b715c4/packages/cdai/src/components/IdeNavigation/Navigation/_IdeNavigation.scss#L47)

**Comment by lee-chase** [comment]
Approving, however, @chrisbygrave we will need to locate an appropriate designer for approval/review before it could move towards release.

They would need to publish the component https://pages.github.ibm.com/cdai-design/pal/ or on a similar Pal site.

**Review by lee-chase — APPROVED** [review]
LGTM

**Pull request merged**